### PR TITLE
Update MQTT Agent as submodule

### PR DESCRIPTION
**Pull request by divekarshubham**
Description
-----------

* Bring in MQTT agent as a submodule
* Delete old agent files
* Add Agent interface functions to OTA demo helpers

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: demos/common/ota_demo_helpers/ota_demo_helpers.c [hunk 1]
```diff
@@ -208,3 +239,137 @@ void SubscriptionManager_RemoveCallback( const char * pTopicFilter,
...
+        freeCommandStructMutex = xSemaphoreCreateCounting( MQTT_COMMAND_CONTEXTS_POOL_SIZE, MQTT_COMMAND_CONTEXTS_POOL_SIZE );
+        initStatus = SEMAPHORE_INITIALIZED;
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+Command_t * Agent_GetCommand( uint32_t blockTimeMs )
+{
+    Command_t * structToUse = NULL;
```
[the remark this answers is not in the document]
Comment (divekarshubham) [reply demos/common/ota_demo_helpers/ota_demo_helpers.c]: Changed to `xCommand` in https://github.com/aws/amazon-freertos/pull/3074/commits/44b852026ff8e5df6f73db06f284223ca142fbc1

File: demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c [hunk 2]
```diff
@@ -39,14 +39,17 @@
...
+#include "mqtt_agent.h"
 
 /* Includes helpers for managing MQTT subscriptions. */
 #include "subscription_manager.h"
 
 /* Header include. */
 #include "agent_message.h"
 
+/* Include coreMQTT-Agent configs. */
+#include "core_mqtt_agent_config.h"
```
[the remark this answers is not in the document]
Comment (divekarshubham) [reply demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c]: Updated in https://github.com/aws/amazon-freertos/pull/3074/commits/44b852026ff8e5df6f73db06f284223ca142fbc1

File: demos/common/ota_demo_helpers/ota_demo_helpers.c [hunk 3]
```diff
@@ -53,6 +63,27 @@ typedef struct SubscriptionManagerRecord
  */
 static SubscriptionManagerRecord_t callbackRecordList[ MAX_SUBSCRIPTION_CALLBACK_RECORDS ] = { 0 };
 
+struct AgentMessageContext
```
[the remark this answers is not in the document]
Comment (divekarshubham) [reply demos/common/ota_demo_helpers/ota_demo_helpers.c:66]: Thanks, updated in https://github.com/aws/amazon-freertos/pull/3074/commits/44b852026ff8e5df6f73db06f284223ca142fbc1

File: demos/common/ota_demo_helpers/ota_demo_helpers.c [hunk 4]
```diff
@@ -208,3 +239,137 @@ void SubscriptionManager_RemoveCallback( const char * pTopicFilter,
...
+    {
+        memset( ( void * ) xCommandStructurePool, 0x00, sizeof( xCommandStructurePool ) );
+        xFreeCommandStructMutex = xSemaphoreCreateCounting( MQTT_COMMAND_CONTEXTS_POOL_SIZE, MQTT_COMMAND_CONTEXTS_POOL_SIZE );
+        ucInitStatus = SEMAPHORE_INITIALIZED;
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+Command_t * Agent_GetCommand( uint32_t ulBlockTimeMs )
```
Comment (muneebahmed10) [inline demos/common/ota_demo_helpers/ota_demo_helpers.c:299]: Would it make more sense to use the latest queue based implementation that's in the [Labs repo](https://github.com/FreeRTOS/Lab-Project-coreMQTT-Agent/blob/main/lib/FreeRTOS/mqtt-agent-interface/freertos_command_pool.c)? It did fix a race condition that resulted from using !NONE to indicate structs in use.

File: demos/common/ota_demo_helpers/CMakeLists.txt [hunk 5]
```diff
@@ -18,4 +18,5 @@ afr_module_dependencies(
     ${AFR_CURRENT_MODULE}
     PUBLIC
         AFR::core_mqtt
```
[the remark this answers is not in the document]
Comment (divekarshubham) [reply demos/common/ota_demo_helpers/CMakeLists.txt]: Removed in https://github.com/aws/amazon-freertos/pull/3074/commits/e7522195647af79251d5600b03849fb58762efce

File: demos/common/ota_demo_helpers/ota_demo_helpers.c (hunk 4 above)
Comment (divekarshubham) [reply demos/common/ota_demo_helpers/ota_demo_helpers.c]: Thanks refactored the agent interface and updated with the latest code as in the Labs repo.
https://github.com/aws/amazon-freertos/pull/3074/commits/b7ccb0bc15dd6f100decba126832437cea387557

File: demos/common/ota_demo_helpers/ota_demo_helpers.c [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*
- * AWS IoT Device SDK for Embedded C V202009.00
+ * FreeRTOS V202012.00
```
Comment (pvyawaha) [inline demos/common/ota_demo_helpers/ota_demo_helpers.c:2]: FreeRTOS coding standard should  be followed here, can be in another PR

File: vendors/espressif/boards/esp32/aws_tests/config_files/aws_iot_network_config.h [hunk 7]
```diff
@@ -52,6 +52,6 @@
  *
  */
 
-#define configENABLED_NETWORKS      ( AWSIOT_NETWORK_TYPE_WIFI | AWSIOT_NETWORK_TYPE_BLE )
+#define configENABLED_NETWORKS      ( AWSIOT_NETWORK_TYPE_WIFI )
```
Comment (pvyawaha) [inline vendors/espressif/boards/esp32/aws_tests/config_files/aws_iot_network_config.h:55]: Why is this change required?

File: vendors/espressif/boards/esp32/aws_tests/config_files/ota_config.h [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+ *
+ *  This configuration parameter is sent with data requests and represents the maximum number of
+ *  data blocks the service will send in response. The maximum limit for this must be calculated
+ *  from the maximum data response limit (128 KB from service) divided by the block size.
+ *  For example if block size is set as 1 KB then the maximum number of data blocks that we can
+ *  request is 128/1 = 128 blocks. Configure this parameter to this maximum limit or lower based on
+ *  how many data blocks response is expected for each data requests.
+ *  Please note that this must be set larger than zero.
+ */
+#define otaconfigMAX_NUM_BLOCKS_REQUEST        4U
```
Comment (pvyawaha) [inline vendors/espressif/boards/esp32/aws_tests/config_files/ota_config.h:102]: If block size is 4KB make the default number of blocks requested 1

File: vendors/espressif/boards/esp32/aws_tests/config_files/ota_config.h [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+ */
+#define otaconfigMAX_NUM_REQUEST_MOMENTUM      32U
+
+/**
+ * @brief The number of data buffers reserved by the OTA agent.
+ *
+ * This configurations parameter sets the maximum number of static data buffers used by
+ * the OTA agent for job and file data blocks received.
+ */
+#define otaconfigMAX_NUM_OTA_DATA_BUFFERS      4U
```
Comment (pvyawaha) [inline vendors/espressif/boards/esp32/aws_tests/config_files/ota_config.h:118]: Keep default to 2 if the number of blocks are reduced to 1

File: demos/common/ota_demo_helpers/ota_demo_helpers.c (hunk 6 above)
Comment (divekarshubham) [reply demos/common/ota_demo_helpers/ota_demo_helpers.c:2]: Thanks, will create a separate PR for this.

File: vendors/espressif/boards/esp32/aws_tests/config_files/aws_iot_network_config.h (hunk 7 above)
Comment (divekarshubham) [reply vendors/espressif/boards/esp32/aws_tests/config_files/aws_iot_network_config.h:55]: For the tests, espressif does not have enough stack to enable both HTTP and BLE together and throws compile time error.

File: vendors/espressif/boards/esp32/aws_tests/config_files/ota_config.h (hunk 9 above)
Comment (divekarshubham) [reply vendors/espressif/boards/esp32/aws_tests/config_files/ota_config.h]: Updated in https://github.com/aws/amazon-freertos/pull/3074/commits/c124ba9b539acb863af043caf3dd3e11c997e62b

File: vendors/espressif/boards/esp32/aws_tests/config_files/ota_config.h (hunk 8 above)
Comment (divekarshubham) [reply vendors/espressif/boards/esp32/aws_tests/config_files/ota_config.h]: Updated in https://github.com/aws/amazon-freertos/pull/3074/commits/c124ba9b539acb863af043caf3dd3e11c997e62b

**Review by pvyawaha — APPROVED** [review]
